sm-core/src/main/java/com/salesmanager/core/business/modules/cms/category/CategoryFileManagerImpl.java

```
@@ -0,0 +1,135 @@
+  }
+
+  public void setUploadImage(CategoryImagePut uploadImage) {
+    this.uploadImage = uploadImage;
+  }
+
+  public CategoryImageGet getGetImage() {
+    return getImage;
+  }
+
+  public void setGetImage(CategoryImageGet getImage) {
+    this.getImage = getImage;
+  }
+
+  public void setRemoveImage(CategoryImageRemove removeImage) {
+    this.removeImage = removeImage;
+  }
+
+
+  public void addCategoryImage(CategoryImage categoryImage, ImageContentFile contentImage)
+      throws ServiceException {
+
+
+    try {
+
+      /** copy to input stream **/
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      // Fake code simulating the copy
+      // You can generally do better with nio if you need...
+      // And please, unlike me, do something about the Exceptions :D
+      byte[] buffer = new byte[1024];
+      int len;
+      while ((len = contentImage.getFile().read(buffer)) > -1) {
+        baos.write(buffer, 0, len);
+      }
+      baos.flush();
+
+      // Open new InputStreams using the recorded bytes
+      // Can be repeated as many times as you wish
+      InputStream is1 = new ByteArrayInputStream(baos.toByteArray());
+      InputStream is2 = new ByteArrayInputStream(baos.toByteArray());
+
+      BufferedImage bufferedImage = ImageIO.read(is2);
+
+
+      if (bufferedImage == null) {
+        LOGGER.error("Cannot read image format for " + categoryImage.getCategoryImage());
+        throw new Exception("Cannot read image format " + categoryImage.getCategoryImage());
+      }
+
+      // contentImage.setBufferedImage(bufferedImage);
+      contentImage.setFile(is1);
+
+
+      // upload original -- L
+      contentImage.setFileContentType(FileContentType.CATEGORY);
+      uploadImage.addCategoryImage(categoryImage, contentImage);
+
+
+    } catch (Exception e) {
+      throw new ServiceException(e);
+    } finally {
+      try {
+        categoryImage.getImage().close();
+      } catch (Exception ignore) {
```

we should not hide such exception, please log it.

---

ok, will handle that.